File: LinearDS/A. Простой стек.py
class SimpleStack:
    def __init__(self):
        self.data = []
 
    def push(self, n):
        self.data.append(n)
 
    def pop(self):
        el = self.data[-1]
        self.data.pop(-1)
        return el
 
    def back(self):
        return self.data[-1]
 
    def size(self):
        return len(self.data)
 
    def clear(self):
        self.data.clear()
 
 
stack = SimpleStack()
 
f_out = open("output.txt", "w")
 
command = ""
commands = []
with open("input.txt") as file:
    commands = [line.rstrip() for line in file]
 
for cmd in commands:
    data = list(cmd.split())
    command = data[0]
    if command == "push":
        stack.push(int(data[1]))
        f_out.write("ok\n")
    elif command == "pop":
        f_out.write(str(stack.pop())+"\n")
    elif command == "back":
        f_out.write(str(stack.back())+"\n")
    elif command == "size":
        f_out.write(str(stack.size())+"\n")
    elif command == "clear":
        stack.clear()
        f_out.write("ok\n")
    else:
        break
 
f_out.write("bye")
f_out.close()
exit()

File: LinearDS/A. Простой стек.cpp
#include <iostream>
#include <vector>

using namespace std;

//ifstream cin("input.txt");
//ofstream cout("output.txt");

struct Stack {
	vector<int>st;
	int ind = -1;

	void push(int x) {
		if (st.size() == ind + 1)
			st.push_back(x);
		else
			st[ind + 1] = x;
		ind++;
		cout << "ok\n";
	}

	void pop() {
		cout << st[ind] << endl;
		ind--;
	}

	void back() {
		cout << st[ind] << "\n";
	}

	void size() {
		cout << ind + 1 << endl;
	}

	void clear() {
		cout << "ok\n";
		ind = -1;
	}

	void exxit() {
		cout << "bye";
		exit(0);
	}
};

int main() {
	Stack st;
	int n; cin >> n;
	for (int i = 0; i < n; i++) {
		string s; cin >> s;
		if (s == "push") {
			int x; cin >> x;
			st.push(x);
		}
		if (s == "pop")
			st.pop();
		if (s == "back") 
			st.back();
		if (s == "size")
			st.size();
		if (s == "clear")
			st.clear();
		if (s == "exit")
			st.exxit();
	}
}
